### PR TITLE
Add types for @mapbox/rehype-prism

### PR DESCRIPTION
**Pull request by remcohaszing**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mapbox__rehype-prism/tslint.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
+        "no-outside-dependencies": false
```
Comment (remcohaszing) [inline types/mapbox__rehype-prism/tslint.json:4]: I don’t understand why, but this created the following error:

```
$ npm test mapbox__rehype-prism

> definitely-typed@0.0.3 test
> node --require source-map-support/register node_modules/@definitelytyped/dtslint/ types mapbox__rehype-prism

dtslint@0.0.127
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.9.0

YOUR TYPESCRIPT VERSION: 4.9.0-dev.20220905

Please only submit bug reports when using the officially supported version.

=============
Error: /home/remco/Projects/DefinitelyTyped/node_modules/buffer/index.d.ts:1:1
ERROR: 1:1  no-outside-dependencies  File /home/remco/Projects/DefinitelyTyped/node_modules/buffer/index.d.ts comes from a `node_modules` but is not declared in this type's `package.json`.  See: https://github.com/microsoft/DefinitelyTyped-tools/blob/master/packages/dtslint/docs/no-outside-dependencies.md

    at testTypesVersion (/home/remco/Projects/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:235:11)
    at runTests (/home/remco/Projects/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:197:7)
    at main (/home/remco/Projects/DefinitelyTyped/node_modules/@definitelytyped/dtslint/src/index.ts:83:5)
```

File: types/mapbox__rehype-prism/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "@definitelytyped/dtslint/dt.json",
+    "rules": {
```
Comment (sheetalkamat) [inline types/mapbox__rehype-prism/tslint.json:3]: Please remove this and fix error. You cannot disable rules in new types

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/mapbox__rehype-prism/tslint.json (hunk 1 above)
Comment (remcohaszing) [reply types/mapbox__rehype-prism/tslint.json:4]: I fixed it upstream in https://github.com/vfile/vfile/pull/82

**Review by sheetalkamat — APPROVED** [review]

**Comment by remcohaszing** [comment]
Ready to merge

**Pull request merged**